### ISSUE TITLE
Assets Handles

### ISSUE THREAD

**Issue by slyedoc** [NiklasEi/bevy_asset_loader#5]
I am new to bevy and ran into a few issues today and both came back to be something with bevy_asset_loader, though maybe I am missing something.  Note that I am forked and using bevy_main, (no rev)

1.  Assets Unloaded

So after adding a few *.wav files to an AssetCollection, I would have an issue where the state change state on load never happening, and the asset collections never got added as resources so panic.  Funny enough though if I ran in --release they would.  After some poking around got some info from  Res<LoadingAssetHandles<Assets>>

AudioSource Collection - Not release
```
"bevy_asset::handle::Handle<bevy_kira_audio::source::AudioSource>" - NotLoaded
"bevy_asset::handle::Handle<bevy_kira_audio::source::AudioSource>" - Loading
"bevy_asset::handle::Handle<bevy_kira_audio::source::AudioSource>" - Loading
"bevy_asset::handle::Handle<bevy_kira_audio::source::AudioSource>" - Unloaded
"bevy_asset::handle::Handle<bevy_kira_audio::source::AudioSource>" - Unloaded
...
```
Texture Collection- Not release
```
"bevy_asset::handle::Handle<bevy_render::texture::texture::Texture>" - NotLoaded
"bevy_asset::handle::Handle<bevy_render::texture::texture::Texture>" - Loading
"bevy_asset::handle::Handle<bevy_render::texture::texture::Texture>" - Loading
"bevy_asset::handle::Handle<bevy_render::texture::texture::Texture>" - Unloaded
"bevy_asset::handle::Handle<bevy_render::texture::texture::Texture>" - Unloaded
...
```
Font Collection - Not release - Works
```
"bevy_asset::handle::Handle<bevy_text::font::Font>" - Loading
"bevy_asset::handle::Handle<bevy_text::font::Font>" - Loaded
```

With Release - All 3 works
```
"bevy_asset::handle::Handle<bevy_render::texture::texture::Texture>" - NotLoaded
"bevy_asset::handle::Handle<bevy_kira_audio::source::AudioSource>" - NotLoaded
"bevy_asset::handle::Handle<bevy_text::font::Font>" - Loading
"bevy_asset::handle::Handle<bevy_text::font::Font>" - Loaded
"bevy_asset::handle::Handle<bevy_kira_audio::source::AudioSource>" - Loaded
"bevy_asset::handle::Handle<bevy_render::texture::texture::Texture>" - Loaded
```

Wtf was going on with Asset LoadState.  It was at this point I ran across [bevy 2347]https://github.com/bevyengine/bevy/issues/2347)

Related?

2. Audio Handle Issue

I was working on adding auto with your bevy_kira_audio, (thanks for all you contribute btw)  while I couldn't get the sounds after the first few frames. I keep thinking I was using bevy_kira_audio wrong, and after refactored my code like 3 times, I finally tried removing bevy_asset_loader from the picture, and boom it works.

I set up a [toy example](https://github.com/slyedoc/bevy-kira-example/blob/main/src/main.rs) to show the problem, am I missing something?

Thanks

**Comment by NiklasEi** [comment]
It looks like I forgot to merge the bug fix of v0.2.1 back to `bevy_main`, sorry about that. And yes, you found the correct Bevy issue where this bug was found.
The problem was, that I wasn't keeping strong handles during the loading of the assets. That means Bevy will start unloading them again. In debug mode your code can be too slow to use the handles while they are loaded and assets don't appear at all while in release, the loading is usually quicker and thus the bug can have less effect.

I will merge `main` back into `bevy_main`. When you pull the changes into your fork this issue should be fixed.

**Comment by slyedoc** [comment]
... I should have realized that, thanks.
